### ISSUE TITLE
pre-install gzip to avoid busybox-gzip via dependency

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-installer#135]
We are aiming at self hosting, i.e. where a resulting rockstor install can, with minimum friction, build another installer, or in the case where this issue has arisen, facilitate the building of a rockstor rpm. However when we attempt to install the required 'rpmbuild' program we are faced with:
```
zypper install --no-recommends rpm-build                  
...

Problem: the to be installed rpm-build-4.14.3-150300.46.1.x86_64 requires 'gzip', but this requirement cannot be provided
  not installable providers: gzip-1.10-7.1.x86_64[Leap_15_4]
                   gzip-1.10-150200.10.1.x86_64[repo-sle-update]
 Solution 1: deinstallation of busybox-gzip-1.35.0-150400.4.3.14.noarch
 Solution 2: do not ask to install a solvable providing rpmbuild
 Solution 3: break rpm-build-4.14.3-150300.46.1.x86_64 by ignoring some of its dependencies

Choose from above solutions by number or cancel [1/2/3/c/d/?] (c): 1
Resolving dependencies...
Resolving package dependencies...

The following 7 NEW packages are going to be installed:
  bzip2 dwz gettext-tools gzip patch python-rpm-macros rpm-build

The following package is going to be REMOVED:
  busybox-gzip

7 new packages to install, 1 to remove.
Overall download size: 1.9 MiB. Already cached: 0 B. After the operation, additional 6.1 MiB will be used.
Continue? [y/n/v/...? shows all options] (y):
```

With the following investigatory command indicating the dependencies that have auto-installed busybox-gzip
```
zypper remove busybox-gzip
...

The following 14 packages are going to be REMOVED:
  busybox-gzip dracut dracut-kiwi-lib dracut-kiwi-oem-dump dracut-kiwi-oem-repart dracut-mkinitrd-deprecated kernel-default-5.14.21-150400.24.41.1
  kernel-default-5.14.21-150400.24.46.1 kexec-tools kmod lvm2 mdadm suse-module-tools udev

14 packages to remove.
After the operation, 400.2 MiB will be freed.
```
Which is obviously a no-go.

However we can successfully choose option 1 in both the original rpm-build install and the following more focused replacement:
```
zypper in --no-recommends gzip
...
Problem: the installed busybox-gzip-1.35.0-150400.4.3.14.noarch conflicts with 'gzip' provided by the to be installed gzip-1.10-150200.10.1.x86_64
 Solution 1: deinstallation of busybox-gzip-1.35.0-150400.4.3.14.noarch
 Solution 2: do not install gzip-1.10-150200.10.1.x86_64

Choose from above solutions by number or cancel [1/2/c/d/?] (c): 1
Resolving dependencies...
Resolving package dependencies...

The following NEW package is going to be installed:
  gzip

The following package is going to be REMOVED:
  busybox-gzip

1 new package to install, 1 to remove.
Overall download size: 140.3 KiB. Already cached: 0 B. After the operation, additional 307.8 KiB will be used.
Continue? [y/n/v/...? shows all options] (y):
```

We have a similar scenario, with the proposed gzip pre-install, in an upstream here:
https://opendev.org/openstack/diskimage-builder/commit/c091227f4edf9f64d47537a18a2fc54e1721df78

**Comment by Hooverdan96** [comment]
Based on a side conversation with @phillxnet  and @FroggyFlox, I am adding additional observations that I encountered during the build process, which I faced while testing the ability to build Rockstor on OpenSUSE Tumbleweed, which proves to be tricky due to it being a rolling release that continues to evolve. Hence, having the build instructions as self-contained as possible should insulate from some of the TW quirks, especially during this transitional phase of gradually getting to more current releases of the underlying building blocks like python 3.x, django, etc.
It appears that the kiwi packages automatically pull in `busybox` packages for its own dependencies if there is nothing explicitly specified beforehand. It doesn’t matter whether the non-busybox versions are installed on the host system (this probably also applies to using the `buildbox` approach, though I have not verified that).
After reading some more in the kiwi docs and looking at the `gzip` issue mentioned above, I needed to find a place in the kiwi install where I “pre-emptively” install `gawk`, `grep` and `xz`.
Adding it to the bootstrap packages seems to do the trick, and kiwi does not reach for busybox as its source for these. This means in the kiwi file these entries could be added:

Bootstrap package section:

![image](https://user-images.githubusercontent.com/35113775/222770455-43823ce5-a0d5-4149-904c-85cdeef642c1.png)

    <packages type="bootstrap">
        <package name="ca-certificates"/>
        
        <package name="ca-certificates-cacert"/>
        <package name="ca-certificates-mozilla"/>
        <package name="cracklib-dict-full"/>
        <package name="filesystem"/>
        <package name="gawk"/>
        <package name="glibc-locale"/>  
        <package name="grep"/>
        <package name="openSUSE-release"/>  
        <package name="udev"/>
        <package name="xz"/>
    </packages>




------------


For Tumbleweed specific conversations I want to leave behind this information as well:
Changed this from `dracut` to `kiwi`, thinking it might make a difference (and since in the kiwi docs it implies when using kiwi it essentially uses dracut anyway). It initially didn’t make a difference of course. Only when adjusting the bootstrap packages (as stated above) the behavior changed (i.e. not using busybox-based installations), but I forgot to change it back, and it still worked.
![image](https://user-images.githubusercontent.com/35113775/222769882-977328f3-a1d4-430f-ab64-59262d8c82f1.png)

For anybody trying to (as of today) replicate this with Tumbleweed: there is currently no TW specific Rockstor build in the corresponding repository.
Until that changes (once more urgent issues have been addressed in the Leap 15.x space) pointing to the most recent (in this case 15.4) in the TW profile of the `rockstor.kiwi` file will allow to successfully build.

![image](https://user-images.githubusercontent.com/35113775/222772743-49c4e3ce-f700-4496-8aba-6591833ebaba.png)
